maxfw/core/utils.py

```diff
@@ -0,0 +1,55 @@
+from flask import abort
+from maxfw.utils.image_utils import ImageProcessor
+
+
+def redirect_errors_to_flask(func):
+    """
+    This decorator function will capture all Pythonic errors and return them as flask errors.
+
+    If you are looking to disable this functionality, please remove this decorator from the `apply_transforms()` module
+    under the ImageProcessor class.
+    """
+
+    def inner(*args, **kwargs):
+        try:
+            # run the function
+            return func(*args, **kwargs)
+        except ValueError as ve:
+            if 'specific_message' in str(ve):
+                raise NotImplementedError
+            else:
+                raise NotImplementedError
+        except TypeError as te:
+            raise te
+            # TODO
+        except Exception as e:
+            # on error, return a 400 using the `abort` module in flask
+            if len(str(e)) > 0:
+                # if there is a specific error message, return it
+                abort(400, str(e))
```

Comment:
A security concern: It is perhaps best to use logger to log `e` but leave the abort message more generic. This avoid giving attackers more information.

Reply:
Good point to have a logger for any (flask) errors - not just limited to the image utils..

Reply:
Yes we want to avoid just returning the underlying exception error message in a `400` response.

If it indeed should be a `400` then return a useful message, while perhaps logging the error in the app.

If it's actually an internal error, it's ok to either:
- just let the exception get thrown. Flask will return a `500 Internal Server Error` and log the exception in the app stdout. (in debug mode, the exception is returned in the API response but if debug = False it is not);
- or ... catch the exception, log it in the app and return an explicit `500` with a message